app/core/modules/Tasks/components/TaskHistory.vue

```diff
@@ -69,20 +163,78 @@
                 required: true,
             },
         },
+        inject: ['reload'],
         data() {
             return {
                 statusService: new StatusService(),
                 priorityService: new PriorityService(),
+                taskActivityService: new taskActivityService(),
                 statuses: [],
                 priorities: [],
+                taskCommentService: new TaskCommentService(),
+                userService: new UsersService(),
+                commentMessage: '',
+                users: [],
+                userFilter: '',
+                userNameStart: 0,
+                userNameEnd: 0,
+                showUsers: false,
+                usersTop: 0,
+                usersLeft: 0,
+                scrollTop: 0,
+                commentMessageScrollTop: 0,
+                idComment: false,
+                changeMessageText: null,
+                sort: 'desc',
+                typeActivity: 'all',
+                activity: [],
+                page: 1,
+                canLoad: true,
             };
         },
+        async created() {
+            this.users = await this.userService.getAll();
+            this.activity = (await this.getActivity()).data;
+        },
         async mounted() {
+            window.addEventListener('scroll', this.onScroll);
             this.statuses = await this.statusService.getAll();
-            this.priorities = await this.priorityService.getAll();
+        },
+        beforeDestroy() {
+            window.removeEventListener('scroll', this.onScroll);
+        },
+        computed: {
+            visibleUsers() {
+                return this.users.filter(user => {
+                    return user.full_name.replace(/\s/g, '').toLocaleLowerCase().indexOf(this.userFilter) === 0;
+                });
+            },
+        },
+        watch: {
+            sort(newQuestion, oldQuestion) {
+                this.resetHistory();
+            },
+            typeActivity(newQuestion, oldQuestion) {
+                this.resetHistory();
+            },
         },
         methods: {
             fromNow,
+            async resetHistory() {
+                this.page = 1;
+                this.canLoad = true;
+                this.activity = (await this.getActivity()).data;
+            },
+            async getActivity(dataOptions = {}) {
+                let data = {
```

Comment:
1. Why do you use 'let' here? You don't change this variable further
2. Why do you declare a variable & dedicate memory for it when you don't need it further - only to pass an object as a param to the function?

Reply:
you right, i fix it